### PR TITLE
compiles with rescript@next (11-rc4)

### PR DESCRIPTION
**Pull request by AlexMoutonNoble**

### CODE REVIEW

File: src/@apollo/client/core/ApolloClient__Core_Types.res [hunk 1]
```diff
@@ -180,7 +180,7 @@ module MutationQueryReducersMap = {
     ~safeParse,
   ) =>
     Js.Dict.map(
-      (. mutationQueryReducer) => mutationQueryReducer->MutationQueryReducer.toJs(~safeParse),
+      (. mutationQueryReducer) => mutationQueryReducer->MutationQueryReducer.toJs(~safeParse, ...),
```
Comment (jeddeloh) [inline src/@apollo/client/core/ApolloClient__Core_Types.res:183]: I think this is the only place where new syntax is used that will force the minimum rescript version to 11? I'm not necessarily against it, but wonder if we should consider something like the below instead so we don't have to release a major version.
```
(. mutationQueryReducer) => (. json, options) => mutationQueryReducer->MutationQueryReducer.toJs(~safeParse, json, options),
```

File: src/@apollo/client/core/ApolloClient__Core_ApolloClient.res [hunk 2]
```diff
@@ -313,12 +313,13 @@ module Js_ = {
   ) => Js.Promise.t<FetchResult.Js_.t<'jsData>> = "mutate"
 
   // onClearStore(cb: () => Promise<any>): () => void;
+  // TODO: named arguments to binding? - AxM
```
Comment (jeddeloh) [inline src/@apollo/client/core/ApolloClient__Core_ApolloClient.res:316]: Did you want to discuss this?

**Comment by jeddeloh** [comment]
Thanks much for this! I made a couple comments. I think we'll have to remember to update package.json to allow rescript 11 as well.

I've sort of been waiting for the official migration documentation to come out before tackling this, but wondered if it would be possible for library authors to simply specify `uncurried: false` in bsconfig. At any rate, the set of required changes seem pretty minimal so maybe this is better regardless.

File: src/@apollo/client/core/ApolloClient__Core_ApolloClient.res (hunk 2 above)
Comment (AlexMoutonNoble) [reply src/@apollo/client/core/ApolloClient__Core_ApolloClient.res:320]: Was just curious myself, doesnt need your time. ill remove it.

**Pull request closed without merging**